pkg/chart/chart.go

```diff
@@ -308,22 +306,17 @@ func (c *ChartClient) InitNetClient(details *Details) (HTTPClient, error) {
 		caCertPool = x509.NewCertPool()
 	}
 
-	namespace := os.Getenv("POD_NAMESPACE")
-	if namespace == "" {
-		namespace = defaultNamespace
-	}
-
 	// We grab the specified app repository (for later access to the repo URL, as well as any specified
 	// auth).
-	appRepo, err := c.appRepoClient.KubeappsV1alpha1().AppRepositories(namespace).Get(details.AppRepositoryResourceName, metav1.GetOptions{})
+	appRepo, err := c.appRepoHandler.GetAppRepository(details.AppRepositoryResourceName, c.kubeappsNamespace, "")
```

Comment:
So this (and the calls below) would use `.GetAppRepositoryForServiceAccount(details.AppRepositoryResourceName, c.kubeappsNamespace)`